tests/backup/backup_test.go

```diff
@@ -4600,6 +4600,193 @@ var _ = Describe("{RestoreEncryptedAndNonEncryptedBackups}", func() {
+				nil, orgID, clusterUid, "", "", "", "", ctx)
+			dash.VerifyFatal(err, nil, "Verifying backup creation")
+		})
+
+		Step("Register source and destination cluster for backup on testuser2", func() {
+			log.InfoD("Registering Source and Destination clusters as user2 and verifying the status")
+			ctx, err := backup.GetNonAdminCtx(users[1], "Password1")
+			log.FailOnError(err, "Fetching testuser ctx")
+			CreateSourceAndDestClusters(orgID, "", "", ctx)
+			portworx.UserContextRegisterCluster = users[1]
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster status")
+		})
+
+		Step("Taking backup of applications as testuser2", func() {
+			log.InfoD("Taking backup of applications as user2")
+			//backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+			var backupName string = "test1"
+			backupNames = append(backupNames, backupName)
+			ctx, err := backup.GetNonAdminCtx(users[1], "Password1")
+			log.FailOnError(err, "Fetching testuser ctx")
+			err = CreateBackup(backupName, SourceClusterName, BackupLocationNameUser2, backupLocationUID, []string{bkpNamespaces[0]},
+				nil, orgID, clusterUid, "", "", "", "", ctx)
+			dash.VerifyFatal(err, nil, "Verifying backup creation")
+		})
+
+		Step("Share backup with testuser2 and validate", func() {
+			log.InfoD("Share backup from user1 to user2 and validate")
+			ctx, err := backup.GetNonAdminCtx(users[0], "Password1")
+
+			// Share backup with the user
+			err = ShareBackup(backupNames[0], nil, []string{users[1]}, RestoreAccess, ctx)
+			log.FailOnError(err, "Failed to share backup %s", backupNames[0])
+		})
+
+		Step("Share backup with testuser1 and validate", func() {
+			log.InfoD("Share backup from user2 to user1 and validate")
+			//log.Infof("Sharing backup with user - %s")
+
+			// Get Admin Context - needed to share backup and get backup UID
+			ctx, err := backup.GetNonAdminCtx(users[1], "Password1")
+
+			// Share backup with the user
+			err = ShareBackup(backupNames[0], nil, []string{users[0]}, RestoreAccess, ctx)
+			log.FailOnError(err, "Failed to share backup %s", backupNames[0])
+		})
+	})
+	JustAfterEach(func() {
+
+		// var then variable name then variable type
+		var first string
+
+		// Taking input from user
+		fmt.Scanln(&first)
+		log.InfoD("Deleting the deployed apps after the testcase")
+		for i := 0; i < len(contexts); i++ {
+			opts := make(map[string]bool)
+			opts[SkipClusterScopedObjects] = true
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			err := Inst().S.Destroy(contexts[i], opts)
+			dash.VerifySafely(err, nil, fmt.Sprintf("Verify destroying app %s, Err: %v", taskName, err))
+		}
+		var wg sync.WaitGroup
+
+		// unset the user context
+		UserContextBackupLocation = ""
+		portworx.UserContextRegisterCluster = ""
```

Comment:
Move these lines up. This is the first thing you need to do before anything else.
What if the destroying app task fails and exits?